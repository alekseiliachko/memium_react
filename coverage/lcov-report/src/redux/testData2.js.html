
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/redux/testData2.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">src/redux</a> testData2.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a>
<a name='L109'></a><a href='#L109'>109</a>
<a name='L110'></a><a href='#L110'>110</a>
<a name='L111'></a><a href='#L111'>111</a>
<a name='L112'></a><a href='#L112'>112</a>
<a name='L113'></a><a href='#L113'>113</a>
<a name='L114'></a><a href='#L114'>114</a>
<a name='L115'></a><a href='#L115'>115</a>
<a name='L116'></a><a href='#L116'>116</a>
<a name='L117'></a><a href='#L117'>117</a>
<a name='L118'></a><a href='#L118'>118</a>
<a name='L119'></a><a href='#L119'>119</a>
<a name='L120'></a><a href='#L120'>120</a>
<a name='L121'></a><a href='#L121'>121</a>
<a name='L122'></a><a href='#L122'>122</a>
<a name='L123'></a><a href='#L123'>123</a>
<a name='L124'></a><a href='#L124'>124</a>
<a name='L125'></a><a href='#L125'>125</a>
<a name='L126'></a><a href='#L126'>126</a>
<a name='L127'></a><a href='#L127'>127</a>
<a name='L128'></a><a href='#L128'>128</a>
<a name='L129'></a><a href='#L129'>129</a>
<a name='L130'></a><a href='#L130'>130</a>
<a name='L131'></a><a href='#L131'>131</a>
<a name='L132'></a><a href='#L132'>132</a>
<a name='L133'></a><a href='#L133'>133</a>
<a name='L134'></a><a href='#L134'>134</a>
<a name='L135'></a><a href='#L135'>135</a>
<a name='L136'></a><a href='#L136'>136</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const testData2 = {
  authReducer: {
    username: "alex",
    authState: "SUCCESS",
  },
  userReducer: {
    avatar: "blob:http://localhost:3000/713f9c81-76c4-4f3f-b361-b0da3e8e3c4a",
    blackList: [
      {
        accountId: "32df86b7-872d-45d5-8a51-e217f0bc973d",
        username: "bob",
        name: null,
        bio: "populated",
        imageData: null,
      },
    ],
    details: {
      accountId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
      name: "Круто чел",
      bio: "populated",
      gender: "gender",
      dob: 1615046987793,
    },
    subs: [
      {
        accountId: "22bb3fe5-185c-4c87-8633-1586586506b7",
        username: "zilber",
        name: null,
        bio: "populated",
        imageData: null,
      },
    ],
    likedList: [
      {
        id: "9064e587-3c29-4547-a5a6-51a8149456f8",
        imageUrl: null,
        authorId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        title: "Русские мемчанские",
        description: null,
        date: 1615046988090,
        category: "Anime",
      },
      {
        id: "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        imageUrl: null,
        authorId: "22bb3fe5-185c-4c87-8633-1586586506b7",
        title: "Chem?",
        description: null,
        date: 1615046988090,
        category: "Chemistry",
      },
      {
        id: "5d8e2ff1-21bc-4fe0-936f-851407318748",
        imageUrl: null,
        authorId: "32df86b7-872d-45d5-8a51-e217f0bc973d",
        title: "It?",
        description: null,
        date: 1615046988090,
        category: "IT",
      },
      {
        id: "5b28e638-ef12-45a7-b629-84350cc0c92e",
        imageUrl: null,
        authorId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        title: "Bio?",
        description: null,
        date: 1615047293904,
        category: "IT",
      },
    ],
    avatarLoadingStatus: "LOADED",
    detailsLoadingStatus: "LOADED",
    subsLoadingStatus: "LOADED",
    blackListLoadingStatus: "LOADED",
    likedListLoadingStatus: "LOADED",
    feedLoadingStatus: "LOADED",
    feed: {
      articles: [
        {
          id: "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
          imageUrl: null,
          authorId: "22bb3fe5-185c-4c87-8633-1586586506b7",
          title: "Chem?",
          description: null,
          date: 1615046988090,
          category: "Chemistry",
        },
      ],
      accounts: [
        {
          accountId: "22bb3fe5-185c-4c87-8633-1586586506b7",
          username: "zilber",
          name: null,
          bio: "populated",
          imageData: null,
        },
      ],
    },
    myArticles: [],
    myArticlesLoadingStatus: "NOT_LOADED",
  },
  articleReducer: {
    articleDataById: {
      "5b28e638-ef12-45a7-b629-84350cc0c92e": {
        id: "5b28e638-ef12-45a7-b629-84350cc0c92e",
        imageUrl: null,
        description: null,
        authorId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        title: "Bio?",
        date: 1615047293904,
        category: "IT",
        data:
          "&lt;div&gt;&lt;p&gt;Рассмотрим две &lt;b&gt;штуки&lt;/b&gt;&lt;/p&gt;&lt;p&gt;Первое:&lt;/p&gt;&lt;p&gt;&lt;b&gt;&lt;br&gt;&lt;/b&gt;&lt;/p&gt;&lt;/div&gt;",
      },
    },
    articleIdsByAuthorName: {},
  },
  allUsersReducer: {
    userDataById: {
      "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282": {
        accountId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        name: "Круто чел",
        bio: "populated",
        gender: "gender",
        dob: 1615046987793,
        avatar:
          "blob:http://localhost:3000/591e585d-6418-41be-bc34-1859656565a2",
      },
    },
  },
  searchReducer: {
    articles: [],
    authors: [],
  },
};
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Mar 08 2021 17:46:47 GMT+0300 (Moscow Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    