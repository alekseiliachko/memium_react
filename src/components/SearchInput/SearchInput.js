import React, { useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Button, Input } from "@material-ui/core";
import { useDispatch } from "react-redux";
import { fetchSearchResults } from "../../redux/search/actions";

const useStyles = makeStyles(() => ({
  root: {
    width: "100%",
    display: "flex",
  },
  input: {
    width: "100%",
    marginRight: "50px",
  },
}));

export const SearchInput = () => {
  const [text, setText] = useState("");
  const dispatch = useDispatch();
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Input
        className={classes.input}
        placeholder={"Поиск"}
        value={text}
        onInput={(e) => setText(e.target.value)}
      />
      <Button
        color="primary"
        onClick={() => dispatch(fetchSearchResults(text))}
        variant="contained"
      >
        Искать
      </Button>
    </div>
  );
};
