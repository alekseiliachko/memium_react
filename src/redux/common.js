export const LOADING_STATUS = {
  NOT_LOADED: "NOT_LOADED",
  LOADING: "LOADING",
  LOADED: "LOADED",
};
