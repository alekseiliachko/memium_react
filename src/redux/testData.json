{
  "authReducer": {
    "username": "alex",
    "authState": "SUCCESS"
  },
  "userReducer": {
    "avatar": "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da",
    "blackList": [
      {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "username": "bob",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "details": {
      "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
      "name": "Круто чел",
      "bio": "populated",
      "gender": "gender",
      "dob": 1615046987793
    },
    "subs": [
      {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "username": "zilber",
        "name": null,
        "bio": "populated",
        "imageData": null
      }
    ],
    "likedList": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "description": null,
        "date": 1615046988090,
        "category": "Chemistry"
      },
      {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "description": null,
        "date": 1615046988090,
        "category": "IT"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "avatarLoadingStatus": "LOADED",
    "detailsLoadingStatus": "LOADED",
    "subsLoadingStatus": "LOADED",
    "blackListLoadingStatus": "LOADED",
    "likedListLoadingStatus": "LOADED",
    "feedLoadingStatus": "LOADING",
    "feed": {
      "articles": [],
      "accounts": null
    },
    "myArticles": [
      {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "description": null,
        "date": 1615046988090,
        "category": "Anime"
      },
      {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "description": null,
        "date": 1615047293904,
        "category": "IT"
      }
    ],
    "myArticlesLoadingStatus": "LOADED"
  },
  "articleReducer": {
    "articleDataById": {
      "d3268ef9-bb52-4967-b1ee-14826b9d8a18": {
        "id": "d3268ef9-bb52-4967-b1ee-14826b9d8a18",
        "imageUrl": null,
        "description": null,
        "authorId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "title": "Chem?",
        "date": 1615046988090,
        "category": "Chemistry",
        "data": "yes, and Jews are responsible."
      },
      "5d8e2ff1-21bc-4fe0-936f-851407318748": {
        "id": "5d8e2ff1-21bc-4fe0-936f-851407318748",
        "imageUrl": null,
        "description": null,
        "authorId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "title": "It?",
        "date": 1615046988090,
        "category": "IT",
        "data": "yes, and Jews are responsible."
      },
      "9064e587-3c29-4547-a5a6-51a8149456f8": {
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Русские мемчанские",
        "date": 1615046988090,
        "category": "Anime",
        "data": "yes, and Jews are responsible."
      },
      "5b28e638-ef12-45a7-b629-84350cc0c92e": {
        "id": "5b28e638-ef12-45a7-b629-84350cc0c92e",
        "imageUrl": null,
        "description": null,
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "title": "Bio?",
        "date": 1615047293904,
        "category": "IT",
        "data": "<div><p>Рассмотрим две <b>штуки</b></p><p>Первое:</p><p><b><br></b></p></div>"
      }
    },
    "articleIdsByAuthorName": {}
  },
  "allUsersReducer": {
    "userDataById": {
      "22bb3fe5-185c-4c87-8633-1586586506b7": {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/595ee9e0-fbd4-4712-98e1-1fd759dca517"
      },
      "32df86b7-872d-45d5-8a51-e217f0bc973d": {
        "accountId": "32df86b7-872d-45d5-8a51-e217f0bc973d",
        "name": null,
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/2bd315a7-60e0-41b7-b856-e48559d16d73"
      },
      "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282": {
        "accountId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "name": "Круто чел",
        "bio": "populated",
        "gender": "gender",
        "dob": 1615046987793,
        "avatar": "blob:http://localhost:3000/678d4330-3d93-4470-98f3-4619ed42c932"
      }
    }
  },
  "searchReducer": {
    "articles": [],
    "authors": []
  }
}
