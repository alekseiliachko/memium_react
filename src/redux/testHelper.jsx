import {Provider} from "react-redux";
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import React from "react";

export const getStubStoreData =  () => require('./testData.json');

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

export const withStore = (initialStore) => {
    const store = mockStore(initialStore || getStubStoreData());
    return function Wrapper({ children }) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
}
