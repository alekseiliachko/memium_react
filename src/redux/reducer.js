import { combineReducers, createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { authReducer } from "./auth/reducer";
import { userReducer } from "./user/reducer";
import { articleReducer } from "./article/reducer";
import { allUsersReducer } from "./allUsers/reducer";
import { searchReducer } from "./search/reducer";

const rootState = combineReducers({
  authReducer,
  userReducer,
  articleReducer,
  allUsersReducer,
  searchReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunk];
export function configureStore() {
  const store = createStore(
    rootState,
    composeEnhancers(applyMiddleware(...middleware))
  );
  return store;
}
