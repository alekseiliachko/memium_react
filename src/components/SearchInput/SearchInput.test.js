import React from "react";
import renderer from "react-test-renderer";
import { SearchInput } from "./SearchInput";
import { withStore } from "../../redux/testHelper";

describe("search input", () => {
  it("renders correctly", async () => {
    let tree;
    const Comp = withStore();

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <SearchInput />
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
