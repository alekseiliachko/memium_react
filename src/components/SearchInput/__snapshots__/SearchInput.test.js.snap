// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search input renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-2"
    onClick={[Function]}
    onInput={[Function]}
  >
    <input
      className="MuiInputBase-input MuiInput-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Поиск"
      type="text"
      value=""
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Искать
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
